src/main/java/org/corfudb/infrastructure/LayoutWorkflowManager.java

```diff
@@ -0,0 +1,231 @@
+package org.corfudb.infrastructure;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.exceptions.LayoutModificationException;
+import org.corfudb.runtime.view.Layout;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Allows us to make modifications to a layout.
+ * <p>
+ * Created by zlokhandwala on 10/12/16.
+ */
+@Slf4j
+public class LayoutWorkflowManager {
+
+    /**
+     * Layout Servers to be modified.
+     */
+    private List<String> layoutServers;
```

Comment:
So, instead of keeping each layout component, wouldn't it be better to keep a layout itself? That is, in the constructor, create a clone of the layout to work on? This would be particularly important if the layout changes in the future.